core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -494,6 +499,48 @@ private ManifestFile newDeleteFilesAsManifest() {
     return cachedNewDeleteManifest;
   }
 
+  @Override
+  protected Map<Integer, String> updatePartitionStats() {
+    Map<Integer, String> newPartitionStatsLocations = new HashMap<>();
+
+    int currentSpecID = ops.current().spec().specId();
+
+    /**
+     *  Update Partition Stats for Delete operation
+     */
+    PartitionStatsMap deletePartitionStatsMap = filterManager.getPartitionStatsMap();
+    deletePartitionStatsMap.getAllUpdatedPartitionSpec().forEach(
+        specId -> {
+          if (specId != currentSpecID) {
+            List<PartitionStatsEntry> statsEntries = getOldPartitionStatsBySpec(specId);
+            PartitionsTable.PartitionMap parMap = deletePartitionStatsMap.getPartitionMap(specId);
+            parMap.subtractFromPartitionStatsEntries(statsEntries);
+            OutputFile outputFile = newPartitionStatsFile(specId);
+            newPartitionStatsLocations.put(specId, outputFile.location());
+            writePartitionStatsEntries(parMap.getAllPartitionStatsEntries(), outputFile, specId);
```

Comment:
Why are we not re-using the partition stats from the previous snapshot instead of creating a new one for each spec?

Reply:
We are re-using partition stats from the previous snapshot only. this loop iterate over only specifications that got updated as a part of deleteFile(CharSequence path), deleteFile(DataFile file), deleteFromRowFilter(Expression expr) operation.  deletePartitionStatsMap.getAllUpdatedPartitionSpec() this will return only partition specifications that got updated.